00_basics/sc_dimensions.ipynb

```diff
@@ -0,0 +1,1059 @@
+    "Lets start by implementing a function which will multiply two matrices which also checks if this is a valid operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "purple-syracuse",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> a Function"
+     ]
+    }
+   ],
+   "source": [
+    "~matrixMul = { |a, b|\n",
+    "    if(a[0].size != b.size) { Error(\"matrix dimensions do not match\").throw };\n",
+    "    if(a.every({|i| a[0].size == i.size}).not) {Error(\"input A has non identical dimensions\").throw};\n",
+    "    if(b.every({|i| b[0].size == i.size}).not) {Error(\"input B has non identical dimensions\").throw};\n",
+    "    a.collect { |d| (b * d).sum }\n",
+    "};"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2fdb5f5-9a5c-4cbd-b55f-9d3607b226e3",
+   "metadata": {},
+   "source": [
+    "Lets take a look at the formula \n",
+    "\n",
+    "$$\n",
+    "\\underbrace{\\begin{pmatrix}\n",
+    "    1 & 1 & 1 \\\\\n",
+    "    1 & 1 & 3 \\\\\n",
+    "\\end{pmatrix}}_{A}\n",
+    "\\underbrace{\\begin{pmatrix}\n",
+    "    2\\\\\n",
+    "    1\\\\\n",
+    "    1\\\\\n",
+    "\\end{pmatrix}}_{B}\n",
+    "\\overset{!}{=}\n",
+    "\\underbrace{\\begin{pmatrix}\n",
+    "    4 \\\\\n",
+    "    6 \n",
+    "\\end{pmatrix}}_{C}\n",
+    "$$\n",
+    "\n",
+    "$A$ is a $2 \\times 3$ matrix and $B$ a $3 \\times 1$ matrix, so the resulting matrix $C$ must be a $2 \\times 1$ matrix.\n",
+    "The values can easily be verified if we take a look at the definition of matrix multiplication"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "atomic-criticism",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ [ 4 ], [ 6 ] ]"
+     ]
+    }
+   ],
+   "source": [
+    "~matrixMul.(\n",
+    "    [[1, 1, 1], [1,1,3]],\n",
+    "    [[2],[1],[1]]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "higher-michigan",
+   "metadata": {},
+   "source": [
+    "We see that our function works as intended - although for a proper validation of this we cannot work with examples but need to examine the code.\n",
+    "\n",
+    "We now want to use this function to calculate downmixes in SuperCollider.\n",
+    "To do this we first need to start the server."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "potential-therapist",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> localhost"
+     ]
+    }
+   ],
+   "source": [
+    "s.boot();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0e449612-8fa3-4378-9dc9-8aac94c65690",
+   "metadata": {},
+   "source": [
+    "Lets formalise what we want to do before we start coding it.\n",
+    "\n",
+    "$$\n",
+    "\\begin{aligned}\n",
```

Comment:
I'll check this later, because I can't read this …

Reply:
Fixed some grammar issues.
A Link to  ReviewNB can be found @ https://app.reviewnb.com/capital-G/musikinformatik-sose2021/blob/sc-dim/00_basics/sc_dimensions.ipynb/